File: Terceiro semestre/python/sistema_login_python/usuario_service.py
from banco_dados import ler_usuarios, salvar_usuarios
from usuario import Usuario
from email_service import enviar_email_confirmacao

def cadastrar_usuario(novo_usuario):
    usuarios = ler_usuarios()
    if any(u["email"] == novo_usuario.email for u in usuarios):
        print("Usuário já cadastrado.")
        return False
    usuarios.append(novo_usuario.to_dict())
    salvar_usuarios(usuarios)
    enviar_email_confirmacao(novo_usuario.email)
    return True

def autenticar(usuario, senha):
    usuarios = ler_usuarios()
    for u in usuarios:
        if u["usuario"] == usuario and u["senha"] == senha:
            return u
    return None

def atualizar_usuario(email, nova_senha=None, novo_usuario=None):
    usuarios = ler_usuarios()
    for u in usuarios:
        if u["email"] == email:
            if nova_senha:
                u["senha"] = nova_senha
            if novo_usuario:
                u["usuario"] = novo_usuario
            salvar_usuarios(usuarios)
            return True
    return False

# NOVAS FUNÇÕES ADMINISTRATIVAS

def listar_todos_usuarios():
    return ler_usuarios()
 
def excluir_usuario(email):
    usuarios = ler_usuarios()
    novos_usuarios = [u for u in usuarios if u["email"] != email]
    if len(novos_usuarios) == len(usuarios):
        return False  # Nenhum usuário excluído (não encontrado)
    salvar_usuarios(novos_usuarios)
    return True

File: Terceiro semestre/java/sistema_login_java/UsuarioService.java
import java.io.*;
import java.util.*;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;

public class UsuarioService {
    private static final String CAMINHO_BANCO = "c:/Users/kferraz/Documents/GitHub/Atividades-da-Faculdade/Terceiro semestre/Banco_de_dados_json/banco/BancoUsuarios2.json";
    private static List<Usuario> usuarios = new ArrayList<>();
    private static final ObjectMapper mapper = new ObjectMapper();

    public static void carregarUsuarios() throws IOException {
        File arquivo = new File(CAMINHO_BANCO);
        if (arquivo.exists()) {
            usuarios = mapper.readValue(arquivo, new TypeReference<List<Usuario>>(){});
        }
    }

    public static void salvarUsuarios() throws IOException {
        mapper.writerWithDefaultPrettyPrinter().writeValue(new File(CAMINHO_BANCO), usuarios);
    }

    public static List<Usuario> listarUsuarios() {
        return usuarios;
    }

    public static boolean cadastrarUsuario(Usuario novo) throws IOException {
        for (Usuario u : usuarios) {
            if (u.email.equals(novo.email)) {
                return false;
            }
        }
        usuarios.add(novo);
        salvarUsuarios();
        return true;
    }

    public static Usuario autenticar(String usuario, String senha) {
        for (Usuario u : usuarios) {
            if (u.usuario.equals(usuario) && u.senha.equals(senha)) {
                return u;
            }
        }
        return null;
    }

    public static boolean atualizarUsuario(String email, String novaSenha, String novoUsuario) throws IOException {
        for (Usuario u : usuarios) {
            if (u.email.equals(email)) {
                if (novaSenha != null && !novaSenha.isEmpty()) u.senha = novaSenha;
                if (novoUsuario != null && !novoUsuario.isEmpty()) u.usuario = novoUsuario;
                salvarUsuarios();
                return true;
            }
        }
        return false;
    }

    public static boolean excluirUsuario(String email) throws IOException {
        boolean removido = usuarios.removeIf(u -> u.email.equals(email));
        if (removido) salvarUsuarios();
        return removido;
    }
}
